packages/react/src/components/ImageComponent/ImageComponent.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+import { settings } from 'carbon-components';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ *  sorts media query min-widths order to ensure the
+ * browser returns the proper sources and the specified widths
+ *
+ * @param {Array} sources image sources and min-widths
+ *
+ * @returns {Array} sorted array of sources
+ */
+const sortSources = sources => {
+  return sources.sort((a, b) => (a.minWidth > b.minWidth ? -1 : 1));
+};
+
+/**
+ * renders background image
+ *
+ * @param {object} props props object
+ * @param {object} props.images array of images used for diff breakpoints
+ * @param {string} props.defaultImage default image (usually image for largest breakpoint)
```

Comment:
Is the `defaultImage` necessary? It looks like we're already looping through the array, maybe we could just pull out the largest/smallest, first, or last image, or maybe allow them to specify? Something like this maybe?

```jsx
<Picture
  alt=""
  breakpoints={[{
    breakpoint: 'lg',
    src: 'https://example.com/lg.jpg'
  }, {
    breakpoint: 'xlg',
    src: 'https://example.com/xlg.jpg'
  }, {
    breakpoint: 'md',
    src: 'https://example.com/md.jpg'
  }]}
/>
```

```html
<picture>
  <source srcset="https://example.com/lg.jpg" media="(min-width: 1056px)" />
  <source srcset="https://example.com/xlg.jpg" media="(min-width: 1312px)" />
  <source srcset="https://example.com/md.jpg" media="(min-width: 675px)" />
  <img class="bx--picture__img" src="https://example.com/lg.jpg" alt="" />
</picture>
```

Grabbing the first one would probably be the easiest for now, and we could always expand from there if we need to. What do you think?

Reply:
Perhaps the logic should be in the pre-defined breakpoints, grab those in the order of availability (`lg`, `md`, etc) and if none are available, use the first one?

Reply:
added break points to the logic